Source/Classes/PINRemoteImageManager.m

```diff
@@ -1356,7 +1429,17 @@ - (BOOL)materializeAndCacheObject:(id)object
     }
     
     if (diskData) {
-        [self.cache setObjectOnDisk:diskData forKey:key];
+        // maxAge of 0 means that images should not be stored at all.
+        // There is no HTTP header that can be sent to indicate "infinite". However not setting a value at all, which in
+        // our case is represented by maxAge == nil, effectively means that.
+        if (!(maxAge && [maxAge integerValue] == 0)) {
```

Comment:
Is this more readable (honest question)? (I think it's still correct)

```
if (maxAge && [maxAge integerValue] != 0 && _isTtlCache) {
    [self.cache setObjectOnDisk:diskData forKey:key withAgeLimit:[maxAge integerValue]];
} else if (!maxAge) {
    // unset (nil) maxAge, or a cache that is not _isTtlCache behave as before (will use cache global ageLimit)
    [self.cache setObjectOnDisk:diskData forKey:key];
}
```

Reply:
I dunno? Yes it looks still correct, and the first case is more readable, but it seems harder to reason through the other two possibilities...

Reply:
No I think it is also incorrect in that if `maxAge` is set but `_isTtlCache` is false, the proposal fails to call the old method (does not cache at all)

Reply:
I think this may suggest we should have a simple local variable with a more descriptive name, like:

BOOL neverStoreImage = (maxAge && [maxAge integerValue] == 0);
if (neverStoreImage == NO) {

}

There's probably a better variable name or another way to do it that is more readable still, like you could check:

BOOL storeImageIndefinitely = !maxAge;
if (storeImageIndefinitely || [maxAge integerValue] > 0) {

}

This second one might be the most readable?

Reply:
Excellent suggestion! See the newest edition 👍